File: backend/src/main/java/com/uoh/model/entity/Book.java
package com.uoh.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;

/**
 * 图书表
 * @TableName book
 */
@TableName(value ="book")
@Data
public class Book implements Serializable {
    /**
     * 图书ID
     */
    @TableId(type = IdType.ASSIGN_ID)
    @JsonSerialize(using = ToStringSerializer.class)
    private Long id;

    /**
     * 图书名称
     */
    private String bookName;


    /**
     *图书介绍
     */
    private String bookIntroduction;

    /**
     * 图书封面URL
     */
    private String cover;

    /**
     * 作者
     */
    private String author;

    /**
     * 出版社
     */
    private String publisher;

    /**
     * 出版日期
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm", timezone = "Asia/Shanghai")
    private Date publishDate;

    /**
     * ISBN（可为空，但唯一）
     */
    private String isbn;

    /**
     * 总库存
     */
    private Integer totalStock;

    /**
     * 可借库存
     */
    private Integer availableStock;

    /**
     * 已经借出去的数量
     */
    private Integer borrowedCount;

    /**
     *累计借阅次数
     */
    private Integer totalBorrowedCount;

    /**
     * 图书状态：0下架 1上架
     */
    private Integer status;

    /**
     * 价格
     */
    private BigDecimal price;

    /**
     * 语言
     */
    private String language;

    /**
     * 创建时间
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm", timezone = "Asia/Shanghai")
    private Date createTime;

    /**
     * 更新时间
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm", timezone = "Asia/Shanghai")
    private Date updateTime;

    @TableField(exist = false)
    private static final long serialVersionUID = 1L;
}

File: frontend/admin-panel/src/api/book.js
import { http } from "@/utils/http.js";

/**
 * 新增图书
 * @param {Object} data
 * @param {string} data.bookName - 图书名称
 * @param {string} data.bookIntroduction - 图书简介
 * @param {string} data.cover - 封面URL
 * @param {string} data.author - 作者
 * @param {string} data.publisher - 出版社
 * @param {string} data.publishDate - 出版日期 (ISO字符串)
 * @param {string} data.isbn - ISBN
 * @param {number} data.totalStock - 总库存
 * @param {number} data.price - 价格
 * @param {string} data.language - 语言
 */
export const addBook = (data) => {
    return http.post("/book/add", data);
};

/**
 * 删除图书
 * @param {Object} data
 * @param {number|string} data.id - 图书ID
 */
export const deleteBook = (data) => {
    return http.post("/book/delete", data);
};

/**
 * 更新图书
 * @param {Object} data
 * @param {number|string} data.id - 图书ID (必填)
 * @param {number} data.status - 状态：0下架 1上架
 * @param {Object} [其他字段参照新增接口]
 */
export const updateBook = (data) => {
    return http.post("/book/update", data);
};

/**
 * 分页获取图书列表
 * @param {Object} data
 * @param {number} data.currentPage - 当前页码
 * @param {number} data.pageSize - 每页条数
 * @param {string} [data.bookName] - 图书名称筛选
 * @param {string} [data.author] - 作者筛选
 * @param {number} [data.publishDateType] - 0:之前 1:之后
 */
export const listBook = (data) => {
    return http.post("/book/list/page", data);
};